client/src/lib.rs

```diff
@@ -527,6 +529,8 @@ pub struct Client {
     namespace: String,
     /// If set, attach as the worker build id to relevant calls
     bound_worker_build_id: Option<String>,
+    /// If set, task completions will include the appropriate version stamp
+    use_versioning: bool,
```

Comment:
For someone advocating for making illegal states unrepresentable in code, I would have expected you to use an enum such as:
```
Unversioned,
UnversionedWithBuildId(string),
Versioned(string),
```